engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -155,39 +149,7 @@ public final Type[] allTypes(EnsoContext ctx) {
   public void generateGetters(EnsoLanguage language) {
     if (gettersGenerated) return;
     gettersGenerated = true;
-    var roots = new HashMap<String, RootNode>();
-    if (constructors.size() != 1) {
-      var names = new HashMap<String, List<GetFieldWithMatchNode.GetterPair>>();
-      constructors
-          .values()
-          .forEach(
-              cons -> {
-                Arrays.stream(cons.getFields())
-                    .forEach(
-                        field -> {
-                          var items =
-                              names.computeIfAbsent(field.getName(), (k) -> new ArrayList<>());
-                          items.add(
-                              new GetFieldWithMatchNode.GetterPair(cons, field.getPosition()));
-                        });
-              });
-      names.forEach(
-          (name, fields) -> {
-            roots.put(
-                name,
-                new GetFieldWithMatchNode(
-                    language, name, this, fields.toArray(new GetFieldWithMatchNode.GetterPair[0])));
-          });
-    } else {
-      var cons = constructors.values().toArray(AtomConstructor[]::new)[0];
-      Arrays.stream(cons.getFields())
-          .forEach(
-              field -> {
-                roots.put(
-                    field.getName(),
-                    new GetFieldNode(language, field.getPosition(), this, field.getName()));
-              });
-    }
+    var roots = AtomConstructor.collectFieldAccessors(language, this);
```

Comment:
Introduction of `collectFieldAccessors` function allows use to hide `GetFieldWithMatchNode` and `GetFieldNode` and make them _package private_.